boltdesignsystem/bolt
# DS-137 Add icon support in Menu component and a new display prop for Share component

## Jira

https://pegadigitalit.atlassian.net/browse/DS-137

## Summary

Adds icon support in Menu component and a new `display` prop for Share component.

## Details

1. `icon_before` and `icon_after` added to menu-item.
2. `display` prop added to Share component. `inline`and `menu` are available.
3. Revamped docs for Share component.
4. Revamped docs for Menu component.
5. Fixed display and alignment issues with Icon component.
6. Fixed icon background markup and z-index shenanigan.
7. Added `aria-hidden` to icons.
8. Removed hardcoded icon size in copy to clipboard component. Now icon size matches text size.
9. Changed how `spacing` is handled in menu and menu-item twig, css, and js.
10. Removed `last` logic in menu-item because it's no longer needed.
11. Updated test snaps for a bunch of components which had icon markup in their tests.

## How to test

Run the branch locally and check all the docs for Share, Menu, Copy to clipboard, and Icon. Watch out for any typos or broken demos.
